src/plugins/cloudinary/models/video-source.js

```diff
@@ -198,39 +198,42 @@ class VideoSource extends BaseSource {
     return sources.some((_source) => isSrcEqual(_source, source));
   }
 
-  generateSources() {
-    let isIe = typeof navigator !== 'undefined' && (/MSIE/.test(navigator.userAgent) || /Trident\//.test(navigator.appVersion));
-    let srcs = this.sourceTypes().map((sourceType) => {
-      let src = null;
-      const srcTransformation = this.sourceTransformation()[sourceType] || this.transformation();
-      const format = normalizeFormat(sourceType);
-      let isAdaptive = (['mpd', 'm3u8'].indexOf(format) !== -1);
-      const opts = {};
-      if (srcTransformation) {
-        opts.transformation = Array.isArray(srcTransformation) ? srcTransformation : [srcTransformation];
-      }
-      assign(opts, { resource_type: 'video', format });
-
-      let type = null;
-      let codecTrans = null;
-      let hasCodecSrcTrans = (isKeyInTransformation(opts.transformation, 'video_codec') || isKeyInTransformation(opts.transformation, 'streaming_profile'));
-      [type, codecTrans] = formatToMimeTypeAndTransformation(sourceType);
-      // If user's transformation include video_codec then don't add another video codec to transformation
-      if (codecTrans && !hasCodecSrcTrans) {
-        opts.transformation.push(codecTrans);
-      }
-      if (opts.format === 'auto') {
-        delete opts.format;
-      }
+  generateSource(sourceType) {
+    // Create source transformation array
+    let transformation = this.sourceTransformation()[sourceType] || this.transformation();
+    transformation = Array.isArray(transformation) ? transformation : [transformation];
+    const sourceTypeFormat = normalizeFormat(sourceType);
+    const isAdaptive = (['mpd', 'm3u8'].indexOf(sourceTypeFormat) > -1);
+    const transformationFormat = (transformation.find(t => t && t.format) || {}).format;
+    // Get format from transformation, or from sourceType
```

Comment:
Here we allow to get format from transformation instead of from sourceType, but only for generating the url, isAdaptive is still determined by the sourceType and not by the transformation.